src/Config/Converters/EntityActionConverterFactory.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.ObjectModel;
+
+namespace Azure.DataApiBuilder.Config.Converters;
+
+/// <summary>
+/// Used to convert an <see cref="EntityAction"/> to and from JSON by creating a <see cref="EntityActionConverter"/> if needed.
+/// </summary>
+/// <remarks>
+/// This is needed so we can remove the converter from the options before we deserialize the object to avoid infinite recursion.
+/// </remarks>
+internal class EntityActionConverterFactory : JsonConverterFactory
+{
+    /// <inheritdoc/>
+    public override bool CanConvert(Type typeToConvert)
+    {
+        return typeToConvert.IsAssignableTo(typeof(EntityAction));
+    }
+
+    /// <inheritdoc/>
+    public override JsonConverter? CreateConverter(Type typeToConvert, JsonSerializerOptions options)
+    {
+        return new EntityActionConverter();
+    }
+
+    private class EntityActionConverter : JsonConverter<EntityAction>
+    {
+        public override EntityAction? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.String)
+            {
+                EntityActionOperation op = JsonSerializer.Deserialize<EntityActionOperation>(ref reader, options);
+
+                return new EntityAction(Action: op, Fields: null, Policy: new(Request: null, Database: null));
```

Comment:
Since `Request` and `Database` have their default values as `null` in `Policy` record, we don't need to specify them again here right?